### PR TITLE
DOCS-11719 OneDrive Connector 1.0

### PR DESCRIPTION
**Pull request by valkyrie69**

### CODE REVIEW

File: microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+Some common use cases for OneDrive Connector include:
+
+* Sales Operations
+** When a sales contract comes in through the Salesforce CRM system, upload the contract to OneDrive and collaboratively review the contract with the Deal Desk. You can assign approval tasks, which when approved or rejected, notifies the contract owner.
+** When a new budget proposal or customer invoice is generated, upload the documents to OneDrive and collaborate with the legal team for their review. Assign tasks, which when approved or rejected, notifies the invoice owner.
+** When an opportunity is closed (won), create a folder structure in OneDrive based on a predefined template.
+* Employee onboarding and offboarding 
+** When a new employee starts their job, existing resources can be shared with them via OneDrive. 
+** Create a new folder in OneDrive and share it with the new employee so they can upload documents to it. 
+** When a new document is uploaded to this folder, HR is notified.
```
Comment (dancazacu1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc]: This implies that OneDrive Connector has triggers.
This is not the case with 1.0.0 release.

Comment (valkyrie69) [reply microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc]: @dancazacu1 Are you talking about line 35 only? I assume so. Just want to confirm.

Comment (dancazacu1) [reply microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc]: yes

**Review by dancazacu1 — DISMISSED** [review]

File: microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+* Credentials to access the target resource
+* Anypoint Studio version 7.3.5 or later
+
+== Common Use Cases for the Connector
+
+Some common use cases for OneDrive Connector include:
+
+* Sales Operations
+** When a sales contract comes in through the Salesforce CRM system, upload the contract to OneDrive and collaboratively review the contract with the Deal Desk. You can assign approval tasks, which when approved or rejected, notifies the contract owner.
```
Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc]: notifies > notify

Suggestion: You can assign approval tasks to notify the contract owner when a contract is approved or rejected.

File: microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+** When a new budget proposal or customer invoice is generated, upload the documents to OneDrive and collaborate with the legal team for their review. Assign tasks, which when approved or rejected, notifies the invoice owner.
```
Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc]: Similar rewording suggestion. 

Also, I think "team" is singular.

File: microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+** When an opportunity is closed (won), create a folder structure in OneDrive based on a predefined template.
+* Employee onboarding and offboarding 
+** When a new employee starts their job, existing resources can be shared with them via OneDrive.
```
Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc]: Suggestion: When new employees start their jobs...

File: microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+* Sales Operations
+** When a sales contract comes in through the Salesforce CRM system, upload the contract to OneDrive and collaboratively review the contract with the Deal Desk. You can assign approval tasks, which when approved or rejected, notifies the contract owner.
+** When a new budget proposal or customer invoice is generated, upload the documents to OneDrive and collaborate with the legal team for their review. Assign tasks, which when approved or rejected, notifies the invoice owner.
+** When an opportunity is closed (won), create a folder structure in OneDrive based on a predefined template.
+* Employee onboarding and offboarding 
+** When a new employee starts their job, existing resources can be shared with them via OneDrive. 
+** Create a new folder in OneDrive and share it with the new employee so they can upload documents to it. 
+** After an employee leaves the company, all documents used or uploaded can be moved or deleted (GDPR).
+* Health care +
+Share sensitive patient data only with the departments that are allowed to access the information by using OneDrive to share the files and allow access to the relevant parties.
```
Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc]:
```suggestion
Share sensitive patient data with the departments that are allowed to access the information by using OneDrive to share the files, and allow access to the relevant parties.
```

File: microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+
+== Authentication Types
+
+OneDrive Connector uses the following authentication methods:
+
+* OAuth 2.0 authorization code grant connection type, which is used in apps that are installed on a device to gain access to protected resources.
+* OAuth 2.0 client credentials grant, which uses the identity of the application to access web-hosted resources.
```
Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc:45]:
```suggestion
* OAuth 2.0 client credentials grant, which uses the identity of the app to access web-hosted resources.
```

File: microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+** After an employee leaves the company, all documents used or uploaded can be moved or deleted (GDPR).
+* Health care +
+Share sensitive patient data only with the departments that are allowed to access the information by using OneDrive to share the files and allow access to the relevant parties.
+
+
+== Authentication Types
+
+OneDrive Connector uses the following authentication methods:
+
+* OAuth 2.0 authorization code grant connection type, which is used in apps that are installed on a device to gain access to protected resources.
```
Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/index.adoc]: You have "apps" here and "application" in the next bullet.

File: microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc [hunk 8]
```diff
@@ -0,0 +1,1317 @@
...
+
+
+== Types
+[[ProxyConfiguration]]
+=== Proxy Configuration
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Host a| String | Host where the proxy requests is sent. |  | x
```
Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc]:
```suggestion
| Host a| String | Host where the proxy requests are sent. |  | x
```

File: microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc [hunk 9]
```diff
@@ -0,0 +1,1317 @@
...
+* MICROSOFTONEDRIVE:TIMEOUT 
+* MICROSOFTONEDRIVE:TOO_MANY_REQUESTS 
+* MICROSOFTONEDRIVE:UNAUTHORIZED 
+
+
+[[ListChildren]]
+== List Children
+`<microsoftOneDrive:list-children>`
+
+Retrieves a collection of DriveItems in the children relationship of a DriveItem.
```
Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc]: Maybe "Retrieves a collection of DriveItems from the DriveItem's child relationships."

File: microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc [hunk 10]
```diff
@@ -0,0 +1,1317 @@
...
+
+== Types
+[[ProxyConfiguration]]
+=== Proxy Configuration
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Host a| String | Host where the proxy requests is sent. |  | x
+| Port a| Number | Port where the proxy requests is sent. |  | x
```
Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc]:
```suggestion
| Port a| Number | Port where the proxy requests are sent. |  | x
```

File: microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc [hunk 11]
```diff
@@ -0,0 +1,1317 @@
...
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Parent DriveItem ID a| String |  ID of the folder to copy the selected driveItem to. |  | x
+| Drive Item name a| String |  Name to use when creating the file. |  | x
+| DriveItem Type a| Enumeration, one of:
+
+** FOLDER
+** FILE |  The type of driveItem to be created, it can be either FILE or FOLDER |  FILE |
```
Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc]:
```suggestion
** FILE |  The type of driveItem to create, which can be either FILE or FOLDER |  FILE | 
```

File: microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc [hunk 12]
```diff
@@ -0,0 +1,1317 @@
...
+Executes a copy request to Microsoft OneDrive.
+
+=== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| DriveItem ID a| String | ID of the driveItem to copy. |  | x
+| Parent DriveItem ID a| String | ID of the folder to copy the selected driveItem to. If unspecified, the selected file is copied in the same folder in which it resides. |  |
```
Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc:185]:
```suggestion
| Parent DriveItem ID a| String | ID of the folder to copy the selected driveItem to. If unspecified, the selected file is copied to the same folder in which it resides. |  | 
```

Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc]: Maybe:

...is duplicated in the same folder in which it resides.

...is copied to the same folder in which it resides.

File: microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc [hunk 13]
```diff
@@ -0,0 +1,1317 @@
+= Microsoft OneDrive Module Connector Reference
```
Comment (dhirsch1) [inline microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc:1]: DriveItem in the descriptive text should really be "drive item," but it's up to you whether to expend the effort on this.

Comment (valkyrie69) [reply microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc:1]: It's actually called a driveItem in the Microsoft API documentation.

**Review by dhirsch1 — APPROVED** [review]

**Pull request merged**